### PR TITLE
Ensure that EditableText always shows a cursor

### PR DESCRIPTION
**Pull request by jason-simmons**
Previously EditableText would render a text widget with no cursor if the text
value was empty.

Also adjust the height of the cursor widget to reflect the style's line
height, and update the cursor painting to match.

### CODE REVIEW

File: sky/packages/sky/lib/src/widgets/editable_text.dart [hunk 1]
```diff
@@ -203,21 +204,22 @@ class EditableText extends StatefulComponent {
...
+    if (value.composing.isValid) {
+      TextStyle composingStyle = style.merge(const TextStyle(decoration: underline));
+      text = new StyledText(elements: [
+        style,
+        value.textBefore(value.composing),
+        [composingStyle, value.textInside(value.composing)],
+        value.textAfter(value.composing)
+      ]);
+    } else {
       // TODO(eseidel): This is the wrong height if empty!
```
Comment (Hixie) [inline sky/packages/sky/lib/src/widgets/editable_text.dart]: did you fix this TODO?

**Comment by Hixie** [comment]
LGTM.

File: sky/packages/sky/lib/src/widgets/editable_text.dart (hunk 1 above)
Comment (jason-simmons) [inline sky/packages/sky/lib/src/widgets/editable_text.dart]: If I understand correctly, this refers to the fact that the Text widget will compute a size of (0, 0) if the text is empty.  That hasn't changed.

When the text is empty, the size of the overall EditableText should now match the size of the cursor, which is derived from the font size.

**Comment by abarth** [comment]
Test?

**Comment by jason-simmons** [comment]
Added a test of the cursor blink timer.  Does that look right?

File: sky/unit/test/widget/input_test.dart [hunk 2]
```diff
@@ -57,4 +58,43 @@ void main() {
...
+      );
+    }
+
+    new FakeAsync().run((async) {
+      tester.pumpFrame(builder);
+
+      EditableText editableText = tester.findWidget(
+          (Widget widget) => widget is EditableText);
+      Duration blinkDuration = new Duration(
+          milliseconds: editableText.test_cursorBlinkPeriodMs);
```
Comment (abarth) [inline sky/unit/test/widget/input_test.dart]: Why not have the editableText expose this value as a Duration?  We've been trying to move more things to be expressed in Durations.

**Comment by abarth** [comment]
LGTM

File: sky/unit/test/widget/input_test.dart (hunk 2 above)
Comment (jason-simmons) [inline sky/unit/test/widget/input_test.dart]: Done

**Pull request merged**